js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+            'status': this.parseTransferStatus (this.safeString2 (transfer, 'transact_state', 'state')),
+        };
+    }
+
+    parseAccountId (status) {
+        const statuses = {
+            'MAIN': 'spot',
+            'CONTRACT': 'swap',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    parseTransferStatus (status) {
+        const statuses = {
+            'SUCCESS': 'ok',
+            'FAILED': 'failed',
+            'WAIT': 'pending',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    async withdraw (code, amount, address, tag = undefined, params = {}) {
+        [ tag, params ] = this.handleWithdrawTagAndParams (tag, params);
+        const networks = this.safeValue (this.options, 'networks', {});
+        let network = this.safeString2 (params, 'network', 'chain'); // this line allows the user to specify either ERC20 or ETH
+        network = this.safeString (networks, network, network); // handle ETH > ERC-20 alias
+        this.checkAddress (address);
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        if (tag !== undefined) {
+            address += ':' + tag;
+        }
+        const request = {
+            'currency': currency['id'],
+            'address': address,
+            'amount': amount,
+        };
+        if (network !== undefined) {
+            request['chain'] = network;
+            params = this.omit (params, [ 'network', 'chain' ]);
+        }
+        const response = await this.spot2PrivatePostAssetWithdraw (this.extend (request, params));
+        //
+        //     {
+        //         "code":200,
+        //         "data": {
+        //             "withdrawId":"25fb2831fb6d4fc7aa4094612a26c81d"
+        //         }
+        //     }
+        //
+        const data = this.safeValue (response, 'data', {});
+        return this.parseTransaction (data, currency);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        const [ section, access ] = api;
+        [ path, params ] = this.resolvePath (path, params);
+        let url = undefined;
+        if (section === 'spot') {
+            url = this.urls['api'][section][access] + '/api/' + this.version + '/' + path;
+            let paramsEncoded = '';
+            if (access === 'private') {
+                params['timestamp'] = Date.now ();
+                params['recvWindow'] = this.safeInteger (this.options, 'recvWindow', 5000);
+            }
+            if (Object.keys (params).length) {
+                paramsEncoded = this.urlencode (params);
+                url += '?' + paramsEncoded;
+            }
+            if (access === 'private') {
+                this.checkRequiredCredentials ();
+                const signature = this.hmac (this.encode (paramsEncoded), this.encode (this.secret), 'sha256');
+                url += '&signature=' + signature;
```

Comment:
In PHP this is translated to `&$signature=` (with the $) breaking the signing.

You can do instead:
`url += '&' + 'signature=' + signature;`

Reply:
done !  also, seems other exchanges also have same problem when you search `&signature`  - do they also break? if you have time, you can PR for them too , one or two cases .